api/client/client.go

```diff
@@ -1319,6 +1319,55 @@ func (c *Client) DeleteAllNodes(ctx context.Context, namespace string) error {
 	return trail.FromGRPC(err)
 }
 
+// StreamSessionEvents streams audit events from a given session recording.
+func (c *Client) StreamSessionEvents(ctx context.Context, sessionID string, startIndex int) (chan events.AuditEvent, chan error) {
+	request := &proto.StreamSessionEventsRequest{
+		SessionID:  sessionID,
+		StartIndex: int32(startIndex),
+	}
+
+	ch := make(chan events.AuditEvent)
+
+	stream, err := c.grpc.StreamSessionEvents(ctx, request)
+	if err != nil {
+		e := make(chan error)
+		go func() {
+			e <- trace.Wrap(err)
+			close(ch)
+		}()
+
+		return ch, e
+	}
+
+	e := make(chan error)
+
+	go func() {
+	outer:
+		for {
+			oneOf, err := stream.Recv()
+			if err != nil {
+				if err != io.EOF {
+					e <- trace.Wrap(trail.FromGRPC(err))
+				}
+
```

Comment:
if this unknown error, should we log it and continue to catch?

Reply:
Hmm, that does seem somewhat reasonable but my issue with that is that clients would get potentially dropped events without them knowing if they don't keep a close watch on logs. Also most stream errors from my understanding are fatal and will result in the stream shutting down anyway.

Reply:
Responding here for clarity, I did consider this but I strongly prefer separating case-OK and case-ERR code via a select. We could go either way with this but I do this that my current solution ends up being cleaner in the long run.